### PR TITLE
CAMEL-19050: add DHIS2 component

### PR DESCRIPTION
**Pull request by claudemamo**
# Description

Add DHIS2 Camel Component.

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

**Comment by davsclaus** [comment]
Thanks for the contribution. 

However all the source code must use the ASF license header. So either you need to change this in the PR or we cannot accept the PR to ASF.

Also it would be good to write a bit more in the .doc file what dhis2 is. Just a few lines to explain it to users that do not know.

File: components/camel-dhis2/LICENSE [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Copyright (c) 2004-2022, University of Oslo
```
Comment (oscerd) [inline components/camel-dhis2/LICENSE:1]: The code must be provided with an ASF License compliant license: https://www.apache.org/legal/resolved.html#category-a

As far as I see this project has been licensed as BSD 3-Clause

This copyright of University of Oslo shouldn't be there.

File: components/camel-dhis2/camel-dhis2-api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        <version>4.0.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>camel-dhis2-api</artifactId>
+    <name>Camel DHIS2 API</name>
+    <description>Camel DHIS2 API</description>
+    <packaging>jar</packaging>
+
+    <properties>
+        <firstVersion>3.21.0</firstVersion>
```
Comment (oscerd) [inline components/camel-dhis2/camel-dhis2-api/pom.xml:19]: This should be 4.0.0-SNAPSHOT

**Review by oscerd — CHANGES_REQUESTED** [review]
The code needs to have the ASF License header in all the files.

File: components/camel-dhis2/camel-dhis2-api/pom.xml (hunk 2 above)
Comment (oscerd) [reply components/camel-dhis2/camel-dhis2-api/pom.xml:19]: In that case no, it should be 3.21.0, but only after 3.21.0 will be release, so for the moment it should be 4.0.0-SNAPSHOT, until we release

**Review by oscerd — APPROVED** [review]

**Review by oscerd — APPROVED** [review]
LGTM, I'll try to merge it this week.

**Comment by oscerd** [comment]
Trying to merge this but it's failing with

[ERROR] Failed to execute goal on project camel-dhis2: Could not resolve dependencies for project org.apache.camel:camel-dhis2:jar:4.0.0-SNAPSHOT: Could not find artifact org.apache.camel:camel-dhis2-api:jar:javadoc:4.0.0-SNAPSHOT in apache-public (https://repository.apache.org/content/groups/public/)

**Comment by oscerd** [comment]
I removed the javadoc camel-dhis2-api jar, tests were passing. Merged and pushed. Thanks

**Pull request closed without merging**